docs/using-barbican-kms-plugin.md

```diff
@@ -0,0 +1,96 @@
+# OpenStack Barbican KMS Plugin
+Kubernetes Supports providers for Encrypting secret data at REST. From Kubernetes v1.10 support for KMS encryption provider is also added. KMS encryption provider encrypt using a data encryption key (DEK), The DEKs are encrypted with a key encryption key (KEK) that is stored and managed in a remote KMS. The KMS provider uses gRPC to communicate with a specific KMS plugin.
```

Comment:
Kubernetes supports using KMS providers to encry data in etcd since v1.10.  KMS encryption provider encrypts the data using a data encryption key (DEK); a new DEK is generated for each encryption. The DEKs are encrypted with a key encryption key (KEK) that is stored and managed in a remote KMS. The KMS provider uses gRPC to communicate with a specific KMS plugin. The KMS plugin, which is implemented as a gRPC server and deployed on the same host(s) as the Kubernetes master(s), is responsible for all communication with the remote KMS.

Reply:
@lingxiankong here KMS encryption provider means https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#providers, you can say that kubernetes does the job of encrypting data with DEK and barbican plugin does the job of encrypting the DEK's with KEK generated and  stored with barbican.
Hope, It is bit clear now, yea i agree it looks confusing, will make it clear in the doc.